File: jira-frontend/src/models/Task.js
// src/models/Task.js

export default class Task {
    constructor(id, title, description, status, projectID, createdAt, updatedAt) {
      this.id = id;
      this.title = title;
      this.description = description;
      this.status = status;
      this.projectID = projectID;
      this.createdAt = createdAt;
      this.updatedAt = updatedAt;
    }
  }
  

File: backend/models/task.go
package models

import "gorm.io/gorm"

type Task struct {
	gorm.Model          // id, created/updated/deleted at
	Title       string  `gorm:"not null"` // title (required)
	Description string  // description
	Status      string  `gorm:"not null"` // status (required)
	ProjectID   uint    // associated project id (required)
	Project     Project `gorm:"foreignKey:ProjectID"` // defines task-project relationship in database
}
